### ISSUE TITLE
DPL-572 Bioscan - API endpoint to fetch control locations for destination barcode

### ISSUE THREAD

**Issue by andrewsparkes** [sanger/sequencescape#3705]
**User story**
Create an API endpoint for the Beckman method to call so it knows where to place controls / miss a sample for the creation of the Lysate plate. Minimally this just takes the Lysate destination plate barcode and returns the well locations where the positive and negative controls should go, and the rest is a stamp. User will have created Lysate plate in LIMS by this point (pending state). This may be added to the Sequencescape API or created as a new mini-service.

**Who are the primary contacts for this story**
Beckman
Scott T

**Who is the nominated tester for UAT**
TBC

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] Beckman to retrieve control well locations for a given destination Lysate plate barcode

Implementation
- [x] Beckman calls POST to new Lighthouse Service endpoint with user/barcode/robot `{"user": "user1", "robot": "robot1", "barcode": "SQPP-1234-A"}`
- [x] Lighthouse, calls  existing v2 GET endpoint in SS for plate, with barcode, to fetch the sample info
- [x] If plate barcode doesn't exist, return 500 with error (**returning 400**)
- [x] In Lighthouse, Check plate purpose of plate is "LBSN-96 Lysate". If not, fail 500 with error (**returning 400**)
- [x] Loop over samples, checking controls exist and storing locations.  If missing controls, fail 500 with error (**returning 400**)
- [x] Successful response 200 with data `{"barcode": "SQPP-1234-A", "positive_control": "H12", "negative_control": "D5"}`
- [ ] DEPRECATED for MVP - Record `lystate_creation_event' in Events WH with data: user/ robot/ barcode/ locations
- [ ] DEPRECATED for MVP - (Add event type to Events WH repo and confluence. with description for DB. Run migration on deployment)
- [x] Go through Todo's, refactoring where possible
- [x] Unit tests
- [ ] NOT NEEDED - Integration Suite
- [x] Pick and return only pcr positive and pcr negative controls; not other types, e.g., lysate negative

**Dependencies**
This story is blocked by the following dependencies:
NA

**References**
This story has a non-blocking relationship with:
[DPL-568](https://github.com/sanger/limber/issues/1196)
Integration test - https://github.com/sanger/General-Backlog-Items/issues/233
Limber pipeline configuration - https://github.com/sanger/sequencescape/issues/3703
Sequencescape pipeline configuration - https://github.com/sanger/sequencescape/issues/3703
Limber bed verification for Stock to Lysate plate - https://github.com/sanger/limber/issues/1199

**Additional context**
Bioscan process map [here](https://lucid.app/lucidchart/b21a6020-df09-4be1-af69-26403ed467a1/edit?viewport_loc=1860%2C1619%2C1885%2C1065%2Cwi47cHrbc6ZW&invitationId=inv_44a43cc4-479b-4e19-924c-2a0182f4255c)
Beckman API call architecture [here](https://lucid.app/lucidchart/e2747c56-9e72-4a95-a586-9a7add40c319/edit?viewport_loc=291%2C41%2C2265%2C1191%2C0_0&invitationId=inv_4221d5f9-4d6c-4646-896e-ac1b5b57c2c9)

**Comment by emrojo** [comment]
From slack conversation Scott 24 Mar 14:45: Beckman are able to handle the randomisation of both positive and negative control coordinates with one exception:
Where positive_control= H1 AND negative_control = G1.
Any other combination works (including either one of these conditions being true)

**Comment by harrietc52** [comment]
To create a plate in SS:
- start SS server
- Go to Studies/ Create Sample Manifest
- Fill in fields ()
- Upload
- Call SS API `http://127.0.0.1:3000/api/v2/labware?filter[barcode]=<barcode>&include=purpose,receptacles.aliquots.sample`
- This should return plate and sample info

**Comment by yoldas** [comment]
The following is not needed for MVP but it may be necessary in the future.

Check if the Lysate Event creation to the Event WH is still required.

If Event message required:
- Create mapping of user/ robot/ barcode/ control locations to the event warehouse structure: subject type/ role type etc. See Event WH confluence links in story for help. (e.g. user/ robot/ barcode/ control_locations are each a subject)
- Add event type to Events WH repo and create/ update confluence documents. with description for DB. Run migration on deployment
- After sucessfully parsing Sequencescape's response, create the message to send to the Events WH (`lighthouse/classes/messages/warehouse_messages.py` might be useful - see how other events are created in the code for possible reuse)
- Send the `lystate_creation_event' to Events WH with data: user/ robot/ barcode/ control locations
